src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php

```diff
@@ -119,25 +119,12 @@ public function getKey()
 
     public function addConfiguration(NodeDefinition $node)
     {
-        $node->fixXmlConfig('user_provider');
-        $builder = $node
-            ->beforeNormalization()
-                ->ifTrue(function ($v) { return isset($v['key']); })
-                ->then(function ($v) {
-                    if (isset($v['secret'])) {
-                        throw new \LogicException('Cannot set both key and secret options for remember_me, use only secret instead.');
-                    }
-
-                    @trigger_error('remember_me.key is deprecated since version 2.8 and will be removed in 3.0. Use remember_me.secret instead.', E_USER_DEPRECATED);
-
-                    $v['secret'] = $v['key'];
-
-                    unset($v['key']);
-                })
-                ->end()
-            ->children();
+        $node
+            ->fixXmlConfig('user_provider')
+            ->children()
+        ;
 
-        $builder
+        $builder = $node
```

Comment:
this is wrong. ``->scalarNode()`` should be code on the return value of ``->children()``, not on the node itself (which does not have such method)